crates/rpc/rpc-types/src/eth/engine/payload.rs

```diff
@@ -412,14 +424,83 @@ pub enum PayloadValidationError {
     /// Thrown when a new payload contains a wrong block number.
     #[error("invalid block number")]
     InvalidBlockNumber,
+    /// Thrown when a new payload contains a wrong block hash.
+    #[error("invalid block hash: {hash:?}")]
+    InvalidBlockHash { hash: H256 },
     /// Thrown when a new payload contains a wrong state root
-    #[error("invalid merkle root (remote: {remote:?} local: {local:?})")]
+    #[error("invalid merkle root: (remote: {remote:?} local: {local:?})")]
     InvalidStateRoot {
         /// The state root of the payload we received from remote (CL)
         remote: H256,
         /// The state root of the payload that we computed locally.
         local: H256,
     },
+    /// Thrown when some other error occured while processing payload
+    #[error("{0}")]
+    Other(String),
+}
+
+impl Serialize for PayloadValidationError {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for PayloadValidationError {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct PayloadValidationErrorVisitor;
+        impl<'de> Visitor<'de> for PayloadValidationErrorVisitor {
+            type Value = PayloadValidationError;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                formatter.write_str("Expected type was string")
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
```

Comment:
I'd actually fine with just deserializing this into the string variant to reduce complexity here, because this parsing will become an issue if we add more variants.

we don't actually deserialize this anywhere